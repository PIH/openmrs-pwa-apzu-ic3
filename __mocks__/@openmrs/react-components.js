import types from 'redux-types';
// mock methods

const visitRest = {

  createVisit: (params) => {

    const visit = params.visit;

    if (visit.patient != 'patient_uuid' || visit.visitType != 'visit_type_uuid' || visit.location != 'location_uuid') {
      throw {
        name: 'Mock Exception',
        message: 'Unable to Submit'
      };
    }
    else {
      return;
    }
  }

};

const encounterRest = {

  createEncounter: (params) => {

    if (params.encounter.encounterType === 'invalid_encounter_type') {
      throw {
        name: 'Mock Exception',
        message: 'Unable to Submit'
      };
    }
    else {
      return;
    }
  }

};

const BASIC_TYPES = [
  'FETCH_SUCCEEDED',
  'FETCH_PENDING',
  'SET_SUCCEEDED',
  'SET_REQUESTED',
  'SET_FAILED',
  'FETCH_FAILED',
  'FETCH_REQUESTED',
  'SUCCEEDED',
  'PENDING',
  'FAILED',
  'REQUESTED'
];

const LOGIN_TYPES = {
  LOGIN: types('login', BASIC_TYPES),
  LOGIN_LOCATIONS: types('login_locations', BASIC_TYPES),
  LOGOUT: types('logout', BASIC_TYPES)
}

const SESSION_TYPES = types('session', BASIC_TYPES)


export {
  visitRest,
  encounterRest,
  LOGIN_TYPES,
  SESSION_TYPES
};
