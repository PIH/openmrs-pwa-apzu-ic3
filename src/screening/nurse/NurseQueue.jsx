import React from "react";
import { connect } from "react-redux";
import { push } from 'connected-react-router';
import {selectors} from '@openmrs/react-components';
import ScreeningQueue from "../ScreeningQueue";
import nurseFilters from './nurseFilters';

// TODO can we figure out a better way to do this without passing dispatch all the way through?

let NurseQueue = props => {

  const rowSelectedActionCreators = [
    () => push({
      pathname: '/screening',
      state: {
        queueLink: '/screening/nurse/queue'
      }
    })
  ];

  return (
    <div>
      <ScreeningQueue
        dispatch={props.dispatch}
        filters={[(patient) => !nurseFilters.completed(patient)]}
        rowData={Object.values(props.patients)}
        rowSelectedActionCreators={rowSelectedActionCreators}
        title="Nurse Queue"
      />
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    patients: selectors.getPatientStore(state)
  };
};

export default connect(mapStateToProps)(NurseQueue);
