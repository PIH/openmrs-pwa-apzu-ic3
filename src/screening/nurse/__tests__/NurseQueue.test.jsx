import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { DataGrid, visitActions, patientActions } from '@openmrs/react-components';
import NurseQueue from '../NurseQueue';
import { ACTIVE_VISITS_REP, VISIT_REPRESENTATION } from "../../../constants";

let props, store;
let mountedComponent;

const mockStore = configureMockStore();

const nurseQueue = () => {
  if (!mountedComponent) {
    mountedComponent = mount(
      <Provider store={store}>
        <NurseQueue {...props} />
      </Provider>);
  }
  return mountedComponent;
};

describe('Component: NurseQueue', () => {
  beforeEach(() => {
    props = {
      session: {
        sessionLocation: {
          uuid: 'abc'
        }
      }
    };
    store = mockStore(
      {
        dispatch: {},
        openmrs: {
          session: {
            sessionLocation: {
              uuid: 'abc'
            }
          },
          patients: {
            set: {},
            selected: null
          }
        },

      });
    mountedComponent = undefined;
  });

  it('renders properly', () => {
    expect(toJson(nurseQueue())).toMatchSnapshot();
    expect(nurseQueue().find(DataGrid).length).toBe(1);
    expect(nurseQueue().find(DataGrid).props().rowSelectedActionCreators.length).toBe(2);
    let rowSelectedAction = {
      "pathname": '/screening/nurse/nursePage',
      "state": {
        "queueLink": '/screening/nurse/queue'
      }
    };
    expect(nurseQueue().find(DataGrid).props().rowSelectedActionCreators[1]().payload.args[0]).toEqual(rowSelectedAction);
    expect(store.getActions()).toContainEqual(patientActions.clearSelectedPatient());
    expect(store.getActions()).toContainEqual(visitActions.fetchActiveVisits(props.session.sessionLocation.uuid, ACTIVE_VISITS_REP));
  });

});
