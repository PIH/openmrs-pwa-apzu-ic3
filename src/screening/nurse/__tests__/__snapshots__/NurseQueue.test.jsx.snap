// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: NurseQueue renders properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(NurseQueue)
    session={
      Object {
        "sessionLocation": Object {
          "uuid": "abc",
        },
      }
    }
  >
    <NurseQueue
      dispatch={[Function]}
      patients={Object {}}
      session={
        Object {
          "sessionLocation": Object {
            "uuid": "abc",
          },
        }
      }
    >
      <div>
        <Connect(ScreeningQueue)
          dispatch={[Function]}
          filters={
            Array [
              [Function],
            ]
          }
          rowData={Array []}
          rowSelectedActionCreators={
            Array [
              [Function],
            ]
          }
          title="Nurse Queue"
        >
          <ScreeningQueue
            columnDefs={
              Array [
                Object {
                  "field": "uuid",
                  "headerName": "uuid",
                  "hide": true,
                },
                Object {
                  "autoHeight": true,
                  "cellRenderer": [Function],
                  "cellStyle": Object {
                    "line-height": "26px",
                  },
                  "getQuickFilterText": [Function],
                  "headerName": "Identifier",
                },
                Object {
                  "field": "name.givenName",
                  "headerName": "Given Name",
                },
                Object {
                  "field": "name.familyName",
                  "headerName": "Family Name",
                },
                Object {
                  "field": "gender",
                  "headerName": "Gender",
                },
                Object {
                  "field": "age",
                  "headerName": "Age",
                },
                Object {
                  "field": "address.village",
                  "headerName": "Village",
                },
                Object {
                  "headerName": "Checked-in Time",
                  "valueGetter": [Function],
                },
              ]
            }
            dispatch={[Function]}
            filters={
              Array [
                [Function],
              ]
            }
            rowData={Array []}
            rowSelectedActionCreators={
              Array [
                [Function],
              ]
            }
            session={
              Object {
                "sessionLocation": Object {
                  "uuid": "abc",
                },
              }
            }
            title="Nurse Queue"
          >
            <div>
              <List
                columnDefs={
                  Array [
                    Object {
                      "field": "uuid",
                      "headerName": "uuid",
                      "hide": true,
                    },
                    Object {
                      "autoHeight": true,
                      "cellRenderer": [Function],
                      "cellStyle": Object {
                        "line-height": "26px",
                      },
                      "getQuickFilterText": [Function],
                      "headerName": "Identifier",
                    },
                    Object {
                      "field": "name.givenName",
                      "headerName": "Given Name",
                    },
                    Object {
                      "field": "name.familyName",
                      "headerName": "Family Name",
                    },
                    Object {
                      "field": "gender",
                      "headerName": "Gender",
                    },
                    Object {
                      "field": "age",
                      "headerName": "Age",
                    },
                    Object {
                      "field": "address.village",
                      "headerName": "Village",
                    },
                    Object {
                      "headerName": "Checked-in Time",
                      "valueGetter": [Function],
                    },
                  ]
                }
                delayInterval={60000}
                fetchListActionCreator={[Function]}
                filters={
                  Array [
                    [Function],
                    [Function],
                  ]
                }
                onMountOtherActionCreators={
                  Array [
                    [Function],
                  ]
                }
                optionalFilters={
                  Array [
                    Object {
                      "enabled": false,
                      "filter": [Function],
                      "key": "art",
                      "label": "ART",
                    },
                    Object {
                      "enabled": false,
                      "filter": [Function],
                      "key": "eid",
                      "label": "EID",
                    },
                    Object {
                      "enabled": false,
                      "filter": [Function],
                      "key": "ncd",
                      "label": "NCD",
                    },
                  ]
                }
                rowData={Array []}
                rowSelectedActionCreators={
                  Array [
                    [Function],
                    [Function],
                  ]
                }
                title="Nurse Queue"
              >
                <div>
                  <h3>
                    <Label
                      bsClass="label"
                      bsStyle="default"
                    >
                      <span
                        className="label label-default"
                      >
                        Nurse Queue
                      </span>
                    </Label>
                  </h3>
                  <h3>
                    <Label
                      bsClass="label"
                      bsStyle="default"
                    >
                      <span
                        className="label label-default hidden"
                      />
                    </Label>
                  </h3>
                  <Grid
                    bsClass="container"
                    componentClass="div"
                    fluid={false}
                  >
                    <div
                      className="container"
                    >
                      <Row
                        bsClass="row"
                        componentClass="div"
                      >
                        <div
                          className="row"
                        >
                          <Col
                            bsClass="col"
                            componentClass="div"
                            sm={1}
                          >
                            <div
                              className="col-sm-1"
                            >
                              <Checkbox
                                bsClass="checkbox"
                                disabled={false}
                                inline={false}
                                onChange={[Function]}
                                title=""
                              >
                                <div
                                  className="checkbox"
                                >
                                  <label
                                    title=""
                                  >
                                    <input
                                      disabled={false}
                                      onChange={[Function]}
                                      type="checkbox"
                                    />
                                    ART
                                  </label>
                                </div>
                              </Checkbox>
                            </div>
                          </Col>
                          <Col
                            bsClass="col"
                            componentClass="div"
                            sm={1}
                          >
                            <div
                              className="col-sm-1"
                            >
                              <Checkbox
                                bsClass="checkbox"
                                disabled={false}
                                inline={false}
                                onChange={[Function]}
                                title=""
                              >
                                <div
                                  className="checkbox"
                                >
                                  <label
                                    title=""
                                  >
                                    <input
                                      disabled={false}
                                      onChange={[Function]}
                                      type="checkbox"
                                    />
                                    EID
                                  </label>
                                </div>
                              </Checkbox>
                            </div>
                          </Col>
                          <Col
                            bsClass="col"
                            componentClass="div"
                            sm={1}
                          >
                            <div
                              className="col-sm-1"
                            >
                              <Checkbox
                                bsClass="checkbox"
                                disabled={false}
                                inline={false}
                                onChange={[Function]}
                                title=""
                              >
                                <div
                                  className="checkbox"
                                >
                                  <label
                                    title=""
                                  >
                                    <input
                                      disabled={false}
                                      onChange={[Function]}
                                      type="checkbox"
                                    />
                                    NCD
                                  </label>
                                </div>
                              </Checkbox>
                            </div>
                          </Col>
                        </div>
                      </Row>
                    </div>
                  </Grid>
                  <Connect(DataGrid)
                    columnDefs={
                      Array [
                        Object {
                          "field": "uuid",
                          "headerName": "uuid",
                          "hide": true,
                        },
                        Object {
                          "autoHeight": true,
                          "cellRenderer": [Function],
                          "cellStyle": Object {
                            "line-height": "26px",
                          },
                          "getQuickFilterText": [Function],
                          "headerName": "Identifier",
                        },
                        Object {
                          "field": "name.givenName",
                          "headerName": "Given Name",
                        },
                        Object {
                          "field": "name.familyName",
                          "headerName": "Family Name",
                        },
                        Object {
                          "field": "gender",
                          "headerName": "Gender",
                        },
                        Object {
                          "field": "age",
                          "headerName": "Age",
                        },
                        Object {
                          "field": "address.village",
                          "headerName": "Village",
                        },
                        Object {
                          "headerName": "Checked-in Time",
                          "valueGetter": [Function],
                        },
                      ]
                    }
                    rowData={Array []}
                    rowSelectedActionCreators={
                      Array [
                        [Function],
                        [Function],
                      ]
                    }
                  >
                    <DataGrid
                      columnDefs={
                        Array [
                          Object {
                            "field": "uuid",
                            "headerName": "uuid",
                            "hide": true,
                          },
                          Object {
                            "autoHeight": true,
                            "cellRenderer": [Function],
                            "cellStyle": Object {
                              "line-height": "26px",
                            },
                            "getQuickFilterText": [Function],
                            "headerName": "Identifier",
                          },
                          Object {
                            "field": "name.givenName",
                            "headerName": "Given Name",
                          },
                          Object {
                            "field": "name.familyName",
                            "headerName": "Family Name",
                          },
                          Object {
                            "field": "gender",
                            "headerName": "Gender",
                          },
                          Object {
                            "field": "age",
                            "headerName": "Age",
                          },
                          Object {
                            "field": "address.village",
                            "headerName": "Village",
                          },
                          Object {
                            "headerName": "Checked-in Time",
                            "valueGetter": [Function],
                          },
                        ]
                      }
                      dispatch={[Function]}
                      rowData={Array []}
                      rowSelectedActionCreators={
                        Array [
                          [Function],
                          [Function],
                        ]
                      }
                    >
                      <div
                        className="grid-wrapper"
                        id="omrsGrid"
                      >
                        <label>
                          Search: 
                           
                          <input
                            onChange={[Function]}
                            type="text"
                          />
                        </label>
                        <AgGridReact
                          animateRows={true}
                          columnDefs={
                            Array [
                              Object {
                                "field": "uuid",
                                "headerName": "uuid",
                                "hide": true,
                              },
                              Object {
                                "autoHeight": true,
                                "cellRenderer": [Function],
                                "cellStyle": Object {
                                  "line-height": "26px",
                                },
                                "getQuickFilterText": [Function],
                                "headerName": "Identifier",
                              },
                              Object {
                                "field": "name.givenName",
                                "headerName": "Given Name",
                              },
                              Object {
                                "field": "name.familyName",
                                "headerName": "Family Name",
                              },
                              Object {
                                "field": "gender",
                                "headerName": "Gender",
                              },
                              Object {
                                "field": "age",
                                "headerName": "Age",
                              },
                              Object {
                                "field": "address.village",
                                "headerName": "Village",
                              },
                              Object {
                                "headerName": "Checked-in Time",
                                "valueGetter": [Function],
                              },
                            ]
                          }
                          enableFiltering={true}
                          enableSorting={true}
                          id="omrsGrid"
                          onFilterChanged={[Function]}
                          onGridReady={[Function]}
                          onGridSizeChanged={[Function]}
                          onRowDataChanged={[Function]}
                          onSelectionChanged={[Function]}
                          rowClassRules="rowClassRules"
                          rowData={Array []}
                          rowSelection="single"
                          style="width: 100%; height: 100%;"
                        >
                          <div
                            style={
                              Object {
                                "height": "100%",
                              }
                            }
                          />
                        </AgGridReact>
                      </div>
                    </DataGrid>
                  </Connect(DataGrid)>
                </div>
              </List>
            </div>
          </ScreeningQueue>
        </Connect(ScreeningQueue)>
      </div>
    </NurseQueue>
  </Connect(NurseQueue)>
</Provider>
`;
